src/cache/base.py

```diff
@@ -0,0 +1,102 @@
+from functools import wraps
+from typing import Any, Callable
+from sys import getsizeof
+import pickle
+import time
+
+
+class BaseCache:
+    def __init__(self) -> None:
+        # these specific elements can be change subClasses
+        self.dumper = pickle.dumps
+        self.loader = pickle.loads
```

Comment:
Hm... I think it'll be easier to understand if we have a baseCache abstract class, and and different implementations that build on this.
The problem with having a default implementation is that one can override the dumper but not the loader... and leads to unexpected results ^^

Reply:
👌🏽 a good point there that goes perfectly with the previous comment, updated also !